app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -38,6 +39,12 @@ def destroy
 
     private
 
+    def apply_secure_headers_override
+      override_content_security_policy_directives(
+        form_action: ['"self"', @authorize_form.form_action].compact
```

Comment:
this is a different approach than the SAML controller, I chose to only whitelist the domain for the client_id being provided, so we don't leak additional domains that we allow in headers